### ISSUE TITLE
Timeout not throwing error with status TIMEOUT_ERROR

### ISSUE THREAD

**Issue by PopJoestar** [reduxjs/redux-toolkit#2915]
Hey, PR #2401 shows that `fetchBaseQuery` should throw a `TIMEOUT_ERROR` when time out delay is reached. However, I got an error without status `{"message": "Aborted", "name": "AbortError"}`. I'm using 
`"@reduxjs/toolkit": "^1.9.0"` with` "react-native": "0.70.3"`,  `"react": "18.1.0"`. Here is my base API:

```
export const BaseApi = createApi({
  reducerPath: 'baseApi',
  baseQuery: fetchBaseQuery({
    baseUrl: Config.BASE_URL,
    prepareHeaders: headers => {
      const token = Preferences.getToken();
      headers.set('authorization', `Bearer ${token}`);

      return headers;
    },
    timeout: 1,
  }),
  tagTypes: ['Auth', 'Notification', 'Restaurant'],
  endpoints: () => ({}),
});
```

**Comment by phryneas** [comment]
Hmm, according to our tests at 
https://github.com/reduxjs/redux-toolkit/blob/4c32e29335508e332df8aa925d5c879475e75b69/packages/toolkit/src/query/tests/fetchBaseQuery.test.tsx#L947-L950
you should be getting 
```ts
{
        status: 'TIMEOUT_ERROR',
        error: 'AbortError: The user aborted a request.',
      }
```

Can you create a small reproduction?

**Comment by PopJoestar** [comment]
I will do a small reproduction as soon as I am free, but I think, the reason might be `fetch` in React Native which is, maybe, a little different than `fetch` in a browser

**Comment by mjwvb** [comment]
I have the same problem.

When I wrap fetchBaseQuery, I indeed get `{ status: 'TIMEOUT_ERROR', error: 'AbortError: The user aborted a request.' }`. 
However in the react components, useQuery().error returns `{ name: 'AbortError', message: 'Aborted' }`

**Comment by phryneas** [comment]
That seems very weird. Could one of you provide a small reproduction for that?

**Comment by mjwvb** [comment]
This should reproduce the problem:

state/testApi.js

```js
import { createApi, fetchBaseQuery } from '@reduxjs/toolkit/query/react';

export const testApi = createApi({
    reducerPath: 'test',
    baseQuery: fetchBaseQuery({ baseUrl: 'http://google.nl', timeout: 1 }),
    endpoints: (builder) => ({
        getContent: builder.query({ query: () => 'test' }),
    }),
});

// Add reducer/middleware to the state store
```

React component:
```js
import testApi from 'state/testApi';

const App = (props) => {
    const queryState = testApi.endpoints.getContent.useQuery();
    console.log(queryState.error); // {name: 'AbortError', message: 'Aborted'}
}
```

**Comment by aaronfg** [comment]
Just came across this bug myself using React Native.

The transformErrorResponse on the endpoint shows the "TIMEOUT_ERROR" but the error that actually makes it to the useQuery hook is that "AbortError" / "Aborted" one.

I also find it weird that I the case of a timeout aborting the action, even though the error is seen in the transformErrorResponse, any error returned there will not be used. The "AbortError" will ALWAYS be the error Emre) returned.

**Comment by phryneas** [comment]
I just tried it with the reproduction code above, but it works for me - please check out this CodeSandbox as a starter and see if you can give me a full working reproduction:

https://codesandbox.io/s/hungry-ardinghelli-ysjxy2?file=/src/App.tsx

**Comment by ifigueroap** [comment]
I'm working on an Ionic React project, and after adding a timeout to `fetchBaseQuery`, I'm also getting `{name: 'AbortError', message: 'Aborted'}`. Although that doesn't particularly bothers me, I'd like to ask what is the proper way of catching and handling this error.  We are already using an error boundary, but the exception seems to escape it.

This is the part of our code that, I think, should capture the error:

```
const { showBoundary } = useErrorBoundary();

  const {
    data,
    isError,
    error,
  } = TheAPI.useGetQuery(lastUpdateId);

  useEffect(() => {
    if (isError) {
      showBoundary(error);
    }
  }, [isAlertLocationsDataError, alertLocationsDataError]);
```

How should we catch the timeout error in user code?
Thanks

**Comment by phryneas** [comment]
Honestly, with a reproduction we maybe could just fix this. 
Without it, this issue is going to stay an echo chamber of people who are too lazy to create a reproduction.

**Comment by ifigueroap** [comment]
> Honestly, with a full reproduction we maybe could just fix this. Without it, this issue is going to stay an echo chamber of people who are too lazy to create a reproduction.

Look I really like this library, and I'm very thankful for its existence. That being said, hastily calling your users "lazy" is just, let us say, unprofessional. It's ok if you need a reproduction repo, I get it. But it's not ok to call names.

**Comment by phryneas** [comment]
Excuse me, my wording might have been slightly off, but it comes down to the same base message:  
Without someone putting in the effort to reproduce this in a way that I can see this myself, further people asking about solutions in this thread will not help solve the problem. We leave these issues open in the hopes that someone comes in at some point and says "hey, I've created a reproduction", which helps us fix the problem.  
Instead, I get emails essentially saying "I have this problem too" on a *very* regular basis, and every time I am reminded that I cannot help anyone with that. Honestly, that's extremely frustrating.  
The only way we could avoid this would be to close and lock issues - and that doesn't seem like a good solution either.

So let me rephrase it and ask the same question I already asked twice above in this thread: Could you please create any kind of reproduction that actually shows this issue?

**Comment by ifigueroap** [comment]
Thank you for your reply. I understand the frustration you are mentioning, and I didn't mean to be too harsh in my response. So let's clear the air here :) Also, I'll try to create a minimal repo that reproduces the issue. Hopefully for next week.

Nevertheless, In the code you quoted, what happens if the timeout is really small, like 1 millisecond, will it raise the exception before entering the try-catch? Because in my case the error "disappears" when I increase the timeout value...

**Comment by phryneas** [comment]
Thanks, it would be greatly appreciated!

>  what happens if the timeout is really small, like 1 millisecond, will it raise the exception before entering the try-catch

It would abort the AbortSignal, which would not do anything. The exception is thrown by `fetch` encountering an AbortSignal that is being aborted. 

Try it out:
```js
const controller = new AbortController()
controller.abort()
```
has no effect apart from setting `controller.signal.aborted` to `true`.

**Comment by ifigueroap** [comment]
It took some time, but here I go. I made a small reproduction repo https://github.com/ifigueroap/redux-toolkit-timeout-bug
Inside the `TheAPI.ts` file there is a timeout variable, if set to 10 it yields `ABORT_ERROR` in the code. If set to 1000, it will work without errors.

**Comment by andrejpavlovic** [comment]
It seems that there is a race condition. When a fetch request timeout calls `abort()` it could be that case that `TIMEOUT_ERROR` is returned or `AbortError`.

So here the abort is called:
https://github.com/reduxjs/redux-toolkit/blob/f63b862ab78414cf94d6f3d16ba56c08f4af025c/packages/toolkit/src/query/fetchBaseQuery.ts#L276-L279

And here is the race condition:
https://github.com/reduxjs/redux-toolkit/blob/f63b862ab78414cf94d6f3d16ba56c08f4af025c/packages/toolkit/src/createAsyncThunk.ts#L634-L644

**Comment by n-ii-ma** [comment]
Any update on this?

Still getting `AbortError` on React Native.
